src/HttpRequest.h

```diff
@@ -146,13 +149,38 @@ class HttpRequest: public Http::Message
 
     int imslen;
 
-    Ip::Address client_addr;
+    /// mark this request as initiated by Squid (rather than received on a client connection)
+    void selfInitiated();
```

Comment:
We need to rename this so that it does not look like a test method. I cannot think of a good name, but this option is better than leaving it "as is" and better than makeSelfInitiated() IMO:

```suggestion
    void markAsSelfInitiated();
```

Reply:
Done.